routes/verify.go

```diff
@@ -452,3 +484,380 @@ func (fes *APIServer) sendEmail(email *mail.SGMailV3) {
+	idType := req.PostFormValue("idType")
+	// More specific type of ID
+	idSubType := req.PostFormValue("idSubtype")
+
+	// Country of ID
+	idCountry := req.PostFormValue("idCountry")
+
+	// Identifier on ID - e.g. Driver's license number for DRIVING_LICENSE, Passport number for PASSPORT
+	idNumber := req.PostFormValue("idNumber")
+
+	// customerId maps to the userReference passed when creating the Jumio session. userReference represents Public Key
+	userReference := req.PostFormValue("customerId")
+
+	// Jumio TransactionID
+	jumioTransactionId := req.PostFormValue("jumioIdScanReference")
+
+	// Get Public key bytes and PKID
+	var publicKeyBytes []byte
+	if userReference != "" {
+		publicKeyBytes, _, err = lib.Base58CheckDecode(userReference)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Problem decoding user public key (customerId): %v", err))
+			return
+		}
+	} else {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Public key (customerId) is required"))
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: error getting utxoview: %v", err))
+		return
+	}
+
+	pkid := utxoView.GetPKIDForPublicKey(publicKeyBytes)
+	if pkid == nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: No PKID found for public key: %v", userReference))
+		return
+	}
+
+	tstamp := uint64(time.Now().UnixNano())
+	// Always log the payload in global state for the PKID so we can search for all jumio verification payloads for a given user.
+	pkidTstampnanosKey := GlobalStateKeyForPKIDTstampnanosToJumioTransaction(pkid.PKID, tstamp)
+	if err = fes.GlobalStatePut(pkidTstampnanosKey, payloadBytes); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Error putting jumio callback in global state: %v", err))
+		return
+	}
+
+	var userMetadata *UserMetadata
+	userMetadata, err = fes.getUserMetadataFromGlobalState(userReference)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Error getting user metadata from global state: %v", err))
+		return
+	}
+
+	// If the user is already jumio verified, this is an error and we shouldn't pay them again.
+	if userMetadata.JumioVerified {
+		_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: User already verified: %v", err))
+		return
+	}
+
+	if req.FormValue("idScanStatus") != "SUCCESS" {
+		// This means the verification failed. We've logged the payload in global state above, so we now set JumioReturned
+		// to true and bail.
+		userMetadata.JumioReturned = true
+		if err = fes.putUserMetadataInGlobalState(userMetadata); err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("JumioCallback: Error putting user metdata in global state: %v", err))
+		}
+		return
+	}
+
+	// Make sure this id hasn't been verified before.
+	uniqueJumioKey := GlobalStateKeyForCountryIDDocumentTypeSubTypeDocumentNumber(idCountry, idType, idSubType, idNumber)
```

Comment:
I don't see us storing anything for this key?

Reply:
Yeah I definitely killed this during testing, but I've added it back now. thanks